src/crawlers/browser_crawler.js

```diff
@@ -0,0 +1,523 @@
+ *  Only available if used by the crawler.
+ * @property {?ProxyConfiguration} proxyConfiguration
+ *  A reference to the underlying {@link ProxyConfiguration} class that manages the crawler's proxies.
+ *  Only available if used by the crawler.
+ * @property {BrowserPool} browserPool
+ *  A reference to the underlying `BrowserPool` class that manages the crawler's browsers.
+ *  For more information about it, see the [`browser-pool` module](https://github.com/apify/browser-pool).
+ * @property {AutoscaledPool} autoscaledPool
+ *  A reference to the underlying {@link AutoscaledPool} class that manages the concurrency of the crawler.
+ *  Note that this property is only initialized after calling the {@link CheerioCrawler#run} function.
+ *  You can use it to change the concurrency settings on the fly,
+ *  to pause the crawler by calling {@link AutoscaledPool#pause}
+ *  or to abort it by calling {@link AutoscaledPool#abort}.
+ */
+class BrowserCrawler extends BasicCrawler {
+    static optionsShape = {
+        ...BasicCrawler.optionsShape,
+        // TODO temporary until the API is unified in V2
+        handleRequestFunction: ow.undefined,
+
+        handlePageFunction: ow.function,
+        gotoFunction: ow.optional.function,
+
+        handlePageTimeoutSecs: ow.optional.number,
+        preNavigationHooks: ow.optional.array,
+        postNavigationHooks: ow.optional.array,
+
+        browserPoolOptions: ow.object,
+        sessionPoolOptions: ow.optional.object,
+        persistCookiesPerSession: ow.optional.boolean,
+        useSessionPool: ow.optional.boolean,
+        proxyConfiguration: ow.optional.object.validate(validators.proxyConfiguration),
+    };
+
+    /**
+   * @param {BrowserCrawlerOptions} options
+   * All `BrowserCrawler` parameters are passed via an options object.
+   */
+    constructor(options) {
+        ow(options, 'BrowserCrawlerOptions', ow.object.exactShape(BrowserCrawler.optionsShape));
+        const {
+            handlePageFunction,
+            handlePageTimeoutSecs = 60,
+            gotoFunction,
+            persistCookiesPerSession = true,
+            useSessionPool = true,
+            sessionPoolOptions,
+            proxyConfiguration,
+            browserPoolOptions,
+            preNavigationHooks = [],
+            postNavigationHooks = [],
+            ...basicCrawlerOptions
+        } = options;
+
+        if (!useSessionPool && persistCookiesPerSession) {
+            // @TODO: Maybe we could also automatically set persistCookiesPerSession to false when useSessionPool is false and log warning
+            throw new Error('You cannot use "persistCookiesPerSession" without "useSessionPool" set to true.');
+        }
+
+        super({
+            ...basicCrawlerOptions,
+            useSessionPool: false,
+            handleRequestFunction: (...args) => this._handleRequestFunction(...args),
+            handleRequestTimeoutSecs: handlePageTimeoutSecs * BASIC_CRAWLER_TIMEOUT_MULTIPLIER,
```

Comment:
I updated this to in CheerioCrawler because the timeout multiplier is bad.

```js
handleRequestTimeoutSecs: (requestTimeoutSecs + handlePageTimeoutSecs * 2) + 5,
```